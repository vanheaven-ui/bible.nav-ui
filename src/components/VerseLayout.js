import PropTypes from 'prop-types';

const VerseLayout = ({
  params, handleNext, addFavorite, favoriteStatus, isLoading, isAdding,
}) => (
  <div className="verse-layout-wrap">
    <h2>{`${params.bookName}`}</h2>
    <h3>
      Chapter:
      {' '}
      {params.chapterNum}
      {' '}
      Verse:
      {' '}
      {params.verseID}
      {' '}
      { favoriteStatus && (
        <i className="fas fa-star" style={{ color: '#e27c08', marginLeft: 10 }} />
      )}
    </h3>
    <div>
      { isLoading && <p>Loading...</p> }
      { !isLoading && (
        <>
          <span>{params.verseID}</span>
          {' '}
          {params.text}
          <button type="button" className="move" onClick={e => handleNext(e)}>
            <i className="fas fa-angle-right" />
          </button>
        </>
      )}
    </div>
    { !favoriteStatus && (
      <button
        type="button"
        onClick={() => addFavorite()}
      >
        { isAdding && (
          <span style={{ color: '#555', cursor: 'not-allowed' }}>Adding Favorite...</span>
        )}
        { !isAdding && <span> Add to favourites</span> }
        <i className="fas fa-star" style={{ color: '#e27c08', marginLeft: 10 }} />
      </button>
    )}
  </div>
);

VerseLayout.propTypes = {
  params: PropTypes.instanceOf(Object).isRequired,
  addFavorite: PropTypes.func.isRequired,
  handleNext: PropTypes.func.isRequired,
  favoriteStatus: PropTypes.bool.isRequired,
  isLoading: PropTypes.bool.isRequired,
  isAdding: PropTypes.bool.isRequired,
};

export default VerseLayout;
