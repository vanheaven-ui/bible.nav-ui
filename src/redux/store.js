import { applyMiddleware, createStore } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import thunkMiddleware from 'redux-thunk';
import rootReducer from './reducers';

const middlewareEnhancer = applyMiddleware(thunkMiddleware);
const composedEnhancer = composeWithDevTools(middlewareEnhancer);

const store = createStore(rootReducer, undefined, composedEnhancer);

export default store;
