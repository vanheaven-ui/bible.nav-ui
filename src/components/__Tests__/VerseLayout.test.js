import React from 'react';
import ReactDOM from 'react-dom';
import { Provider } from 'react-redux';
import { render, screen, cleanup } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import rootDiv from '../../testHelper';
import store from '../../redux/store';
import '@testing-library/jest-dom';
import VerseLayout from '../VerseLayout';

describe('VerseLayout Component', () => {
  afterEach(cleanup);

  const params = {
    bookName: 'John',
    chapterNum: '1',
    verse: '1',
    text: 'In the begining...',
  };
  const handleNext = () => '';
  const addFavorite = () => '';
  const favoriteStatus = false;
  const isLoading = false;

  it('renders without crashing', () => {
    ReactDOM.render(
      <VerseLayout
        params={params}
        handleNext={handleNext}
        addFavorite={addFavorite}
        favoriteStatus={favoriteStatus}
        isLoading={isLoading}
      />, rootDiv(),
    );
  });

  it('does not change unexpectedly', () => {
    const tree = renderer.create(
      <VerseLayout
        params={params}
        handleNext={handleNext}
        addFavorite={addFavorite}
        favoriteStatus={favoriteStatus}
        isLoading={isLoading}
      />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the correct wrapper element', () => {
    const { container } = render(
      <VerseLayout
        params={params}
        handleNext={handleNext}
        addFavorite={addFavorite}
        favoriteStatus={favoriteStatus}
        isLoading={isLoading}
      />,
    );
    expect(container.firstChild.classList.contains('verse-layout-wrap')).toBe(true);
  });
});
