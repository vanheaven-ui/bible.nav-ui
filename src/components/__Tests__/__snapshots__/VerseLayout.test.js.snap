// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VerseLayout Component does not change unexpectedly 1`] = `
<div
  className="verse-layout-wrap"
>
  <h2>
    John
  </h2>
  <h3>
    Chapter:
     
    1
     
    Verse:
     
     
  </h3>
  <div>
    <span />
     
    In the begining...
    <button
      className="move"
      onClick={[Function]}
      type="button"
    >
      <i
        className="fas fa-angle-right"
      />
    </button>
  </div>
  <button
    onClick={[Function]}
    type="button"
  >
    <span>
       Add to favourites
    </span>
    <i
      className="fas fa-star"
      style={
        Object {
          "color": "#e27c08",
          "marginLeft": 10,
        }
      }
    />
  </button>
</div>
`;
